pegin/getaddress.go

```diff
@@ -0,0 +1,379 @@
+package pegin
+
+import (
+	"bytes"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+
+	"github.com/btcsuite/btcutil"
+
+	"github.com/btcsuite/btcd/txscript"
+
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+
+	"github.com/btcsuite/btcd/chaincfg"
+
+	"github.com/vulpemventures/go-secp256k1-zkp"
+
+	"github.com/vulpemventures/go-elements/address"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+const (
+	pubKeyLen = 33
+
+	MainNet NetworkType = iota
+	RegtestNet
+)
+
+type NetworkType int
+
+type AddressInfo struct {
+	ClaimScript      string
+	MainChainAddress string
+}
+
+// GetAddressInfo simulates getpeginaddress RPC call which returns mainChain
+// address to which BTC should be sent in BTC network and claim script that is
+// to be used in claimpegin RPC alongside with bitcoinTx and txoutproof
+func GetAddressInfo(
```

Comment:
This is ok for testing, but most likely we will not expose this API "one-shot". Will eventually do in a separate repo/utility package if we need in Go posix environments

Reply:
Done

Reply:
@sekulicd I would rename the function `GetPeginAddress`, GetAddressInfo can be confusing as there is a `getaddressinfo` RPC ?

Reply:
u are right current name is not the happiest one, it can be misleading, i was not happy with GetPeginAddress  since it returns two things mainchain_address and claim_script so i thought to have more general name

Reply:
I suggest to remove it altogether https://github.com/vulpemventures/go-elements/pull/161#discussion_r625215790

Reply:
Done